sklearn/metrics/cluster/_supervised.py

```diff
@@ -888,9 +941,7 @@ def mutual_info_score(labels_true, labels_pred, *, contingency=None):
     log_contingency_nm = np.log(nz_val)
     contingency_nm = nz_val / contingency_sum
     # Don't need to calculate the full outer product, just for non-zeroes
-    outer = pi.take(nzx).astype(np.int64, copy=False) * pj.take(nzy).astype(
-        np.int64, copy=False
-    )
+    outer = pi.take(nzx) * pj.take(nzy)
```

Comment:
interesting

Reply:
Yeah - this is simply to make the contingency matrix allow non-integer entries.